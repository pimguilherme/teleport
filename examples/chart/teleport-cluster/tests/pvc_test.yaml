suite: PersistentVolumeClaim
templates:
  - pvc.yaml
tests:
  - it: creates a PersistentVolumeClaim when chartMode=standalone
    set:
      chartMode: standalone
    asserts:
      - template: pvc.yaml
        hasDocuments:
          count: 1
      - template: pvc.yaml
        isKind:
          of: PersistentVolumeClaim
      - matchSnapshot: {}

  - it: creates a PersistentVolumeClaim when chartMode=custom
    set:
      chartMode: custom
    asserts:
      - template: pvc.yaml
        hasDocuments:
          count: 1
      - template: pvc.yaml
        isKind:
          of: PersistentVolumeClaim
      - matchSnapshot: {}

  - it: creates a PersistentVolumeClaim with values from standalone-customsize.yaml
    values:
      - ../.lint/standalone-customsize.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: PersistentVolumeClaim
      - equal:
          path: spec.resources.requests.storage
          value: 50Gi
      - matchSnapshot: {}

  - it: creates a PersistentVolumeClaim with values from custom-customsize.yaml
    values:
      - ../.lint/custom-customsize.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: PersistentVolumeClaim
      - equal:
          path: spec.resources.requests.storage
          value: 50Gi
      - matchSnapshot: {}

  - it: does not create a PersistentVolumeClaim when chartMode=standalone and existingClaimName is not blank
    set:
      chartMode: standalone
      persistence:
        existingClaimName: test-claim
    asserts:
      - hasDocuments:
          count: 0

  - it: does not create a PersistentVolumeClaim when chartMode=custom and existingClaimName is not blank
    set:
      chartMode: custom
      persistence:
        existingClaimName: test-claim
    asserts:
      - hasDocuments:
          count: 0

  - it: does not create a PersistentVolumeClaim with values from standalone-existingpvc.yaml
    values:
      - ../.lint/standalone-existingpvc.yaml
    asserts:
      - hasDocuments:
          count: 0

  - it: does not create a PersistentVolumeClaim with values from custom-existingpvc.yaml
    values:
      - ../.lint/custom-existingpvc.yaml
    asserts:
      - hasDocuments:
          count: 0

  - it: does not create a PersistentVolumeClaim when chartMode=aws
    set:
      chartMode: aws
    asserts:
      - hasDocuments:
          count: 0

  - it: does not create a PersistentVolumeClaim when chartMode=gcp
    set:
      chartMode: gcp
    asserts:
      - hasDocuments:
          count: 0
