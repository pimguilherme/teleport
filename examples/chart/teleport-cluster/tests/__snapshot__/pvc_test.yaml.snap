creates a PersistentVolumeClaim when chartMode=custom:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
creates a PersistentVolumeClaim when chartMode=standalone:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
creates a PersistentVolumeClaim with values from custom-customsize.yaml:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 50Gi
creates a PersistentVolumeClaim with values from standalone-customsize.yaml:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      accessModes:
      - ReadWriteOnce
      resources:
        requests:
          storage: 50Gi
